lib/services/auth_service.dart

```diff
@@ -40,17 +41,47 @@ class AuthService {
         return false;
       }
       User? user = _databaseService.get(profileId);
+
       try {
-        await login(user!);
-        _authStateStream.add(true);
-        return true;
+        if (user != null) {
+          await _apiService.registerAccessToken(user);
+          _authStateStream.add(true);
+          return true;
+        }
       } catch (_) {
+      }
+
         _authStateStream.add(false);
         return false;
-      }
     }
   }
 
+  Future<User?> loginByQuickConnect(
+      String serverAddress, Function(String) code, CancelToken token) async {
+    final (url, candidates) = await inferServerUrl(serverAddress);
+    if (url == null) {
+      throw Exception('No valid server found on the given url\n'
+          '\nTried the following urls:\n'
+          '${candidates.join('\n-------------\n')}');
+    }
+
+    final user = await _apiService.loginByQuickConnect(
+      url,
+      code,
+      token,
+    );
+
+    if (user == null) {
+      return null;
+    }
+
+    _databaseService.put(user.id! + serverAddress, user);
```

Comment:
serverAddress is the user input. We want to use the actual server address thats used by Jellyflix. The current implementation leads to issues when switching and deleting profiles.